addons/hr_holidays/models/hr_leave_allocation.py

```diff
@@ -222,6 +218,27 @@ def _compute_duration_display(self):
                 else float_round(allocation.number_of_days_display, precision_digits=2)),
                 _('hours') if allocation.type_request_unit == 'hour' else _('days'))
 
+    def _inverse_duration_display(self):
+        for allocation in self:
+            try:
+                value = float(allocation.duration_display.split(' ')[0].replace(',','.'))
+            except ValidationError:
+                raise ValidationError(_("Cannot convert allocated days/hours value : '%s' to float", allocation.duration_display))
+
+            if allocation.type_request_unit:
+                allocation.number_of_days = float(value) if allocation.type_request_unit != 'hour' \
+                    else float(value) /allocation._get_hours_per_day()
+            else:
+                raise ValidationError(_("No allocation value unit defined"))
```

Comment:
This seems fishy and error prone, why would you make an informative field editable, and try to parse it by totally knowing that there is a good chance that it doesn't work (eg: user in French, typo, etc).